lib/v2/screens/send_confirmation/interactor/usecases/send_transaction_use_case.dart

```diff
@@ -2,14 +2,41 @@ import 'package:async/async.dart';
 import 'package:seeds/v2/datasource/local/settings_storage.dart';
 import 'package:seeds/v2/datasource/remote/api/profile_repository.dart';
 import 'package:seeds/v2/datasource/remote/api/send_eos_transaction_repository.dart';
+import 'package:seeds/v2/datasource/remote/model/fiat_rate_model.dart';
+import 'package:seeds/v2/datasource/remote/model/rate_model.dart';
 
 export 'package:async/src/result/result.dart';
 
 class SendTransactionUseCase {
   final SendTransactionRepository _sendTransactionRepository = SendTransactionRepository();
-  final accountName = settingsStorage.accountName;
+  final ProfileRepository _profileRepository = ProfileRepository();
+  final fromAccount = settingsStorage.accountName;
 
-  Future<Result> run(String name, String account, Map<String, dynamic> data) {
-    return _sendTransactionRepository.sendTransaction(name, account, data, accountName);
+  Future<Result> run(String toName, String account, Map<String, dynamic> data) {
+    return _sendTransactionRepository.sendTransaction(toName, account, data, fromAccount).then((Result value) async {
+      if (value.isError) {
+        return value;
+      } else {
+        List<Result> profiles = await getProfileData(data['to'], fromAccount);
+
+        return ValueResult(SendTransactionResponse(profiles, value));
+      }
+    }).catchError((error) {
+      return ErrorResult("Error Sending Transaction");
+    });
+  }
+
+  Future<List<Result>> getProfileData(String toAccount, fromAccount) async {
```

Comment:
need to fetch profiles in order to get the user name and the image

Reply:
I see you made a lot of logic work 👍